extensions/objects/templates/ImageWithRegionsEditor.js

```diff
@@ -192,6 +209,44 @@ oppia.directive('imageWithRegionsEditor', [
             height: (area[1][1] - area[0][1]) * $scope.getImageHeight()
           };
         };
+        var resizeRegion = function() {
+          var labeledRegions = $scope.$parent.value.labeledRegions;
+          var resizedRegion = labeledRegions[$scope.selectedRegion].region;
+          var deltaX = $scope.mouseX - $scope.originalMouseX;
+          var deltaY = $scope.mouseY - $scope.originalMouseY;
+          var x = $scope.originalRectArea.x;
+          var y = $scope.originalRectArea.y;
+          var width = $scope.originalRectArea.width;
+          var height = $scope.originalRectArea.height;
+          if (height - $scope.yDirection * deltaY <= 0 &&
+              !$scope.yDirectionToggled) {
+            $scope.yDirectionToggled = true;
+          } else if (height - $scope.yDirection * deltaY >= 0 &&
+            $scope.yDirectionToggled) {
+            $scope.yDirectionToggled = false;
+          }
+          if ($scope.yDirection === 1) {
+            y += $scope.yDirectionToggled ? height : deltaY;
+          } else if ($scope.yDirection === -1) {
+            y += $scope.yDirectionToggled * (deltaY + height);
+          }
+          if (width - $scope.xDirection * deltaX <= 0 &&
+              !$scope.xDirectionToggled) {
+            $scope.xDirectionToggled = true;
+          } else if (width - $scope.xDirection * deltaX >= 0 &&
+            $scope.xDirectionToggled) {
+            $scope.xDirectionToggled = false;
+          }
+          if ($scope.xDirection === 1) {
+            x += $scope.xDirectionToggled ? width : deltaX;
+          } else if ($scope.xDirection === -1) {
+            x += $scope.xDirectionToggled * (deltaX + width);
+          }
+          height = height - $scope.yDirection * deltaY;
+          width = width - $scope.xDirection * deltaX;
+          resizedRegion.area = regionAreaFromCornerAndDimensions(x, y,
+                                width, height);
+        };
```

Comment:
In this method can you store height - $scope.yDirection * deltaY in a variable like newHeight at the beginning and then just reference it later on? I think it would make it much more readable. Same width width. Also, then you could just use newHeight/newWidth in the call to regionAreaFromCornerAndDimensions rather than reassigning to width and height.